validphys2/src/validphys/covmats.py

```diff
@@ -716,7 +719,22 @@ def pdferr_plus_covmat(dataset, pdf, covmat_t0_considered):
     True
     """
     th = ThPredictionsResult.from_convolution(pdf, dataset)
-    pdf_cov = np.cov(th.error_members, rowvar=True)
+
+    if pdf.error_type == 'replicas':
+        pdf_cov = np.cov(th.error_members, rowvar=True)
+
+    elif pdf.error_type == 'symmhessian':
+        rescale_fac = pdf._rescale_factor()
+        hessian_eigenvectors = th.error_members
+        central_predictions = th.central_value
+
+        # need to subtract the central set which is not the same as the average of the
+        # Hessian eigenvectors.
+        X = hessian_eigenvectors - central_predictions.reshape((central_predictions.shape[0],1))
```

Comment:
This could be something like `central_prediction[:, np.newaxis]` (or `central_prediction[:, None]`)